product_pricelist_extended/models/product_pricelist_item.py

```diff
@@ -23,35 +24,106 @@ class ProductPricelistItem(models.Model):
         column1="pricelist_item_id",
         column2="tmpl_id",
         string="Templates",
-        oldname="tmpl_ids",
     )
     price_categ_id = fields.Many2one(
-        comodel_name="product.price.category",
-        string="Pricing Category",
-        oldname="price_categ",
+        comodel_name="product.price.category", string="Pricing Category"
     )
 
     code_inclusion = fields.Char("Code includes")
     code_exclusion = fields.Char("Code excludes")
 
-    # Inherits upstream usage of deprecated api.one
-    # pylint: disable=W8104
-    @api.one
     @api.depends("price_categ_id", "product_tmpl_ids", "product_ids")
     def _get_pricelist_item_name_price(self):
         super()._get_pricelist_item_name_price()
-        if self.price_categ_id:
-            self.name = _("Price Category: %s") % self.price_categ_id.name
-        elif self.product_tmpl_ids:
-            self.name = ",".join(self.product_tmpl_ids.mapped("name"))
-        elif self.product_ids:
-            self.name = ",".join(
-                [p.display_name.replace("[%s]" % p.code, "") for p in self.product_ids]
-            )
-        suffix = []
-        if self.code_inclusion:
-            suffix.append(_("contains %s") % self.code_inclusion)
-        if self.code_exclusion:
-            suffix.append(_("excludes %s") % self.code_exclusion)
-        if suffix:
-            self.name = self.name + _("(Code %s)") % ", ".join(suffix)
+        for item in self:
+            if item.price_categ_id and item.applied_on == "2_product_category":
+                item.name = _("Price Category: %s") % item.price_categ_id.display_name
+            elif item.product_tmpl_ids and item.applied_on == "1_product":
+                item.name = _("Product: %s") % (
+                    ",".join(item.product_tmpl_ids.mapped("display_name"))
+                )
+            elif item.product_ids and item.applied_on == "0_product_variant":
+                item.name = _("Variant: %s") % (
+                    ",".join(
+                        [
+                            p.with_context(display_default_code=False).name
```

Comment:
this code works after changing `display_name` to `name`

Reply:
Are you sure, I think those fields are different. Maybe in v13 it changed. Anyway, it isn't the point. It should work, stop trying to work around it and find the root cause. This is porting code, so it should be functionally the same as before.

What happens with
`", ".join(item.with_context(display....).product_ids.mapped('display_name'))`

Reply:
I tried it yesterday it also doesn't work, it looks like to do with `display_name` field for product_ids. Atm I'm updating ubuntu to 20.04 and can double-check it afterwards

Reply:
All display_name did was call name_get so just call it directly.

`", ".join([p[1] for p in item.with_context(displ...).product_ids.name_get()])`

Reply:
It still doesn't work, however if I run the `name_get()` manually with 'product_ids' it works. The error kept popping up is "Database fetch misses ids ((<NewId origin=36>,)) and has extra ids ((36,)), may be caused by a type incoherence in a previous request", which is same as before when using `display_name`